### PR TITLE
feat: Add `installationId` in LiveQuery connect

### PR DESCRIPTION
**Pull request by mbfakourii**
## Pull Request

- Report security issues [confidentially](https://github.com/parse-community/Parse-SDK-Flutter/security/policy).
- Any contribution is under this [license](https://github.com/parse-community/Parse-SDK-Flutter/blob/master/LICENSE).
- Link this pull request to an [issue](https://github.com/parse-community/Parse-SDK-Flutter/issues?q=is%3Aissue).

## Issue

Closes: #975

## Approach
add `installationId` in LiveQuery `connect`.

## Tasks

- [x] Add tests
- [ ] Add changes to documentation (guides, repository pages, code comments)

### CODE REVIEW

File: packages/dart/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+## [6.3.0](https://github.com/parse-community/Parse-SDK-Flutter/compare/dart-6.2.0...dart-6.3.0) (2023-10-30)
```
Comment (mtrezza) [inline packages/dart/CHANGELOG.md:1]:
```suggestion
## [6.3.0](https://github.com/parse-community/Parse-SDK-Flutter/compare/dart-6.2.0...dart-6.3.0) (2023-10-31)
```

**Review by mtrezza — COMMENTED** [review]
Could you add a test?

**Comment by mbfakourii** [comment]
> Could you add a test?

These additional items are in a private function, it is also for socket operation and cannot be tested.

**Comment by mtrezza** [comment]
Not sure I understand; why can it not be tested?

**Comment by mbfakourii** [comment]
> Not sure I understand; why do you think it cannot be tested?

Changes in [this](https://github.com/parse-community/Parse-SDK-Flutter/blob/5f445640e2a7bd86e503a3cd9a0cabcfb657e889/packages/dart/lib/src/network/parse_live_query.dart#L321) function. This function is a private function. It cannot be accessed from outside.

**Comment by mtrezza** [comment]
If it not possible to access a private function in tests?

**Comment by mbfakourii** [comment]
> If it not possible to access a private function in tests? How are all the LiveQuery features then tested?

There are currently no tests for `LiveQuery`!

**Comment by mtrezza** [comment]
Interesting. It should be easily possible to test an internal methods.

**Comment by mbfakourii** [comment]
> Interesting. It should be easily possible to test an internal method. Can't you just reference the internal method directly in the test file, mock `channel`, execute the method and check whether the connect message has the installationId set?

I did not test with `mock`, I will try to implement

**Comment by mbfakourii** [comment]
@mtrezza 

In my opinion, it is better to examine the LiveQuery tests in a separate issue, because there is currently no written test, and how to write it should be examined.

**Comment by mbfakourii** [comment]
@mtrezza 

Test added

**Comment by mtrezza** [comment]
Congrats for adding the first live query test!

File: packages/dart/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,9 @@
+## [6.3.0](https://github.com/parse-community/Parse-SDK-Flutter/compare/dart-6.2.0...dart-6.3.0) (2023-10-31)
```
Comment (mtrezza) [inline packages/dart/CHANGELOG.md:1]:
```suggestion
## [6.3.0](https://github.com/parse-community/Parse-SDK-Flutter/compare/dart-6.2.0...dart-6.3.0) (2023-11-11)
```

**Review by mtrezza — APPROVED** [review]
Looks good! Waiting for CI to pass...

**Comment by maravilhosinga** [comment]
This should be optional since, it breaks live query when using lower versions of parse server.
Specially for who uses sashido.io

**Comment by mtrezza** [comment]
@maravilhosinga could you please open a new issue and provide more specifics about what's breaking?

**Pull request merged**